src/service/translationService.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import axios from '@nextcloud/axios'
+import { generateOcsUrl } from '@nextcloud/router'
+import store from '../store/index.js'
+
+const checkTranslationServiceAvailability = function() {
+	/// TODO
+	// const response = await axios.post(generateOcsUrl('taskprocessing/tasktypes'), {
+	// 	appId: 'mail',
+	// })
+
+	const isAvailable = true
+	const languages = [
+		{
+			from: 'en',
+			fromLabel: 'English',
+			to: 'fr',
+			toLabel: 'French',
+		},
+		{
+			from: 'en',
+			fromLabel: 'English',
+			to: 'de',
+			toLabel: 'German',
+		},
+		{
+			from: 'fr',
+			fromLabel: 'French',
+			to: 'en',
+			toLabel: 'English',
+		},
+	]
+
+	store.commit('enableTranslation', isAvailable)
+	store.commit('setTranslationLanguages', languages)
+}
+
+const translateText = async function(text, fromLanguage, toLanguage) {
+	try {
+		const scheduleResponse = await axios.post(generateOcsUrl('taskprocessing/schedule'), {
+			input: {
+				origin_language: fromLanguage ?? null,
+				input: text,
+				target_language: toLanguage,
+			},
+			type: 'core:text2text:translate',
+			appId: 'mail',
+		})
+		const task = scheduleResponse.data.ocs.data.task
+		const getTaskOutput = async (task) => {
```

Comment:
I'm not a big fan of the recursion here, But I'm not sure how to go around it. I tried utilising the `completionExpectedAt` locally but it's unreliable.
Also the task can take really long, maybe we should allows user to reduce the modal and get a notification once its done , the way that NC assistant works ? 
But maybe as a followup

Reply:
This recursion was copied from the text app https://github.com/nextcloud/text/pull/6567, I don't know if there is some better standard nextcloud way of doing it, but it doesn't seem so.

Sure, the modal would be good as a followup

Reply:
sure,I'll dismiss the review until a better approach arrises